runtime/gc_include/ObjectAllocationAPI.hpp

```diff
@@ -168,14 +168,13 @@ class MM_ObjectAllocationAPI
 		if (0 != size) {
 			/* Contiguous Array */
 
-			UDATA scale = ((J9ROMArrayClass*)(arrayClass->romClass))->arrayShape;
 
 #if !defined(J9VM_ENV_DATA64)
-			if (!sizeCheck || (size < ((U_32)J9_MAXIMUM_INDEXABLE_DATA_SIZE >> scale)))
```

Comment:
Dmitri pointed out that this size check must be consistent with the size check below. 

We can not rely on the scale as that will not work for valuetypes

Reply:
Would the following make sense? 

```((U_32)J9_MAXIMUM_INDEXABLE_DATA_SIZE / J9ARRAYCLASS_GET_STRIDE(arrayClass)))```